docs/spec.md

```diff
@@ -4667,8 +4667,8 @@ Given that, `results = exec(schedule, inputs)`, where:
     update_computation(results[:][result_index], updates[:][update_index])`.
   * `updated_results` is a copy of `results` with `results[:][result_index]`
     set to `updated_values[:]`.
-  * If `result_index` is out of bounds for `shape(results[:])`, the behavior
-    is implementation-defined.
+  * If `result_index` is out of bounds for `shape(results[:])`, the update is
+    discarded.
```

Comment:
Same question as in #1541: can we leave a paper trail about why we're making this change?

Reply:
Here is the [reference](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/scatter_expander.cc#L364-L368) for this semantics.

Reply:
1) Why is it that unlike for GatherOp, the write is not clamped but skipped?

2) Also, it's not evident to me that the code in scatter_expander.cc is equivalent to what we have in this PR. E.g. in that code, I see DynamicSlice and DynamicUpdateSlice ops and comments reference the adjusted_start_indices behavior, but that is not at all reflected in the updated spec or the interpreter. Could you elaborate?

Reply:
Gather is clamped because some backends that doesn't implement Gather use DynamicSlice, and OOB indices in DynamicSlice is clamped to provide a uniform behavior across different hardwares.

OOB indices in Scatter on the other hand is skipped because nobody wanted the clamping behavior on Scatter.